lib/src/delegates/asset_grid_drag_selection_aggregator.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:wechat_assets_picker/wechat_assets_picker.dart';
+
+class AssetGridDragSelectionAggregator {
+  AssetGridDragSelectionAggregator({
+    required this.delegate,
+  });
+
+  /// [ChangeNotifier] for asset picker.
+  /// 资源选择器状态保持
+  final DefaultAssetPickerBuilderDelegate delegate;
+
+  /// 拖拽状态
+  /// Drag status
+  bool isInDragging = false;
+
+  /// 边缘自动滚动控制器
+  /// Edge Auto Scrolling Detector. Use to support edge auto scroll when drag position reach the edge of device's screen.
+  EdgeDraggingAutoScroller? _autoScroller;
+
+  // An eyeballed value for a smooth scrolling experience.
+  static const double _kDefaultAutoScrollVelocityScalar = 50;
+
+  /// 起始选择序号
+  /// Item index of the first selected item
+  int initialSelectedIdx = -1;
+
+  /// 拖拽选择 或 拖拽取消选择
+  /// Drag to select or deselect state
+  bool dragSelect = true;
+
+  /// 长按启动拖拽
+  /// Long Press to enable drag and select
+  void onDragStart(
+    BuildContext context,
+    LongPressStartDetails details,
+    int index,
+    AssetEntity entity,
+  ) {
+    isInDragging = true;
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    _autoScroller = EdgeDraggingAutoScroller(
+      scrollableState,
+      velocityScalar: _kDefaultAutoScrollVelocityScalar,
+    );
+
+    initialSelectedIdx = index;
+
+    dragSelect = !delegate.provider.selectedAssets.contains(entity);
+  }
+
+  void onDragUpdate(
+    BuildContext context,
+    LongPressMoveUpdateDetails details,
+    double itemSize,
+    int gridCount,
+    double topPadding,
+  ) {
+    if (!isInDragging) {
+      return;
+    }
+
+    if (dragSelect &&
+        delegate.provider.selectedAssets.length ==
+            delegate.provider.maxAssets) {
```

Comment:
What is the idea of this predicate?

Reply:
This is to block the logic of drag update when the users are in select asset mode and they have already hit the max assets limit.